sdk/nodejs/cmd/run/run.ts

```diff
@@ -342,26 +385,27 @@ ${defaultMessage}`,
         const runProgramSpan = tracing.newSpan("language-runtime.runProgram");
 
         try {
-            const projectRoot = projectRootFromProgramPath(program);
-            const packageObject = packageObjectFromProjectRoot(projectRoot);
-
+            const packageRoot = await npmPackageRootFromProgramPath(program);
+            const packageObject = packageObjectFromProjectRoot(packageRoot);
             let programExport: any;
 
+            // If the user provided an entrypoint, we use that file
+            // relative to the package directory.
+            if (packageObject["main"]) {
+                console.log("Observed Main field.");
+                console.log(`Package Root: ${packageRoot}`);
+                console.log(`Main Field: ${packageObject["main"]}`);
+                program = path.join(packageRoot, packageObject["main"]);
+                console.log(`Final path: ${program}`);
```

Comment:
I'm guessing these log statements are just printed as diagnostics if there are issues?

Reply:
Whoops, those snuck in there today! I'll of course rip these out before whacking merging!

Reply:
They were added for debugging a round of tests earlier today! XD